tools/jit/gen_jit_dispatch.py

```diff
@@ -103,6 +107,9 @@ def jit_type_of(arg):
 
 def from_ivalue(arg, value):
     simple_type = arg['simple_type']
+
```

Comment:
Can you comment this, I have no idea why it is here.

Reply:
Yes, will add a comment in code.  The real reason is that we added `Tensor?[]` for `at::index`, but in code gen we still translate it to TensorList in C++, but mark this type `is_nullable=True` in declarations.yaml. When we recover the types in JIT, we examine `is_nullable`, but there's a difference between normal types and `Tensor?[]`, for normal types we just appending `?` to the end like `Scalar?`,  but we will make `?` in the middle not the end, just the special casing. 

I will try to refactor it to make it more clear